LabController/src/bkr/labcontroller/expire_distros.py

```diff
@@ -137,14 +147,49 @@ def main():
                            'Defaults to http://localhost:8000.')
     parser.add_option('--remove-all', default=False, action='store_true',
                       help='Remove all distros from lab controller.')
+    parser.add_option('--name', default=None,
+                      help='Remove all distros with given name. Use "%" for wildcard.')
+    parser.add_option('--family', default=None,
+                      help='Remove all distros for a given family.')
+    parser.add_option('--arch', default=None,
+                      help='Remove all distros for a given architecture. When set to "all", '
+                           'steps thru each available arch to reduce memory usage.')
     options, args = parser.parse_args()
-    try:
-        check_all_trees(options.ignore_errors,
-                        options.dry_run,
-                        options.lab_controller,
-                        options.remove_all)
-    except KeyboardInterrupt:
-        pass
+    startmsg = str("INFO: expire_distros running with --lab-controller=" + options.lab_controller)
+    for i in range(1,len(sys.argv)):
+        startmsg += ' ' + sys.argv[i]
+    print('%s' % (startmsg))
+    filter = {}
+    if options.name:
+        filter['name'] = options.name
+    if options.family:
+        filter['family'] = options.family
+
+    arch_list = []
+    if options.arch:
+        if options.arch == "all":
+            arch_list = [ "x86_64", "ppc", "ppc64le", "ppc64", "i386", "s390", "s390x", "aarch64", "ia64", "arm", "armhfp" ]
```

Comment:
Question - this list is quite fragile, isn't there an API that can tell us what kind of architectures we have?